src/quo2/components/record_audio/record_audio/view.cljs

```diff
@@ -0,0 +1,473 @@
+                                  :container-style style/send-icon-container}]]]))])
+
+(defn- lock-button []
+  [:f>
+   (fn []
+     (let [translate-x-y (reanimated/use-shared-value 20)
+           opacity (reanimated/use-shared-value 0)
+           connector-opacity (reanimated/use-shared-value 0)
+           width (reanimated/use-shared-value 24)
+           height (reanimated/use-shared-value 12)
+           border-radius-first-half (reanimated/use-shared-value 8)
+           border-radius-second-half (reanimated/use-shared-value 8)
+           start-x-y-animation #(do
+                                  (reanimated/animate-shared-value-with-delay translate-x-y 8 50 :linear 116.66)
+                                  (reanimated/animate-shared-value-with-delay connector-opacity 1 0 :easing1 80)
+                                  (reanimated/animate-shared-value-with-delay width 56 83.33 :easing1 63.33)
+                                  (reanimated/animate-shared-value-with-delay height 56 83.33 :easing1 63.33)
+                                  (reanimated/animate-shared-value-with-delay border-radius-first-half 28 83.33 :easing1 63.33)
+                                  (reanimated/animate-shared-value-with-delay border-radius-second-half 28 83.33 :easing1 63.33))
+           reset-x-y-animation #(do
+                                  (reanimated/animate-shared-value-with-timing translate-x-y 0 100 :linear)
+                                  (reanimated/set-shared-value connector-opacity 0)
+                                  (reanimated/set-shared-value width 24)
+                                  (reanimated/set-shared-value height 12)
+                                  (reanimated/set-shared-value border-radius-first-half 8)
+                                  (reanimated/set-shared-value border-radius-second-half 16))
+           fade-in-animation #(do
+                                (reanimated/animate-shared-value-with-timing translate-x-y 0 220 :linear)
+                                (reanimated/animate-shared-value-with-timing opacity 1 220 :linear))
+           fade-out-animation #(do
+                                 (reanimated/animate-shared-value-with-timing translate-x-y 20 200 :linear)
+                                 (reanimated/animate-shared-value-with-timing opacity 0 200 :linear)
+                                 (reanimated/set-shared-value connector-opacity 0)
+                                 (reanimated/set-shared-value width 24)
+                                 (reanimated/set-shared-value height 12)
+                                 (reanimated/set-shared-value border-radius-first-half 8)
+                                 (reanimated/set-shared-value border-radius-second-half 16))]
+       (effect! #(if @recording? (fade-in-animation) (fade-out-animation)) [@recording?])
+       (effect! #(cond @ready-to-lock? (start-x-y-animation) (and @recording? (not @locked?)) (reset-x-y-animation)) [@ready-to-lock?])
```

Comment:
Could you please reformat this line? This is a bit beyond my comprehension.

Reply:
Done